packages/pdf-worker/src/strategies/ChatTranscript.ts

```diff
@@ -0,0 +1,74 @@
+import moment from 'moment';
+
+import type { Data } from '../types/Data';
+import type { IStrategy } from '../types/IStrategy';
+import exportChatTranscript from '../templates/ChatTranscript';
+import type { ChatTranscriptData, PDFMessage } from '../templates/ChatTranscript';
+
+export class ChatTranscript implements IStrategy {
+	private isNewDay(current: PDFMessage, previous: PDFMessage | undefined): boolean {
+		return !previous || !moment(current.ts).isSame(previous.ts, 'day');
+	}
+
+	private parserMessages(messages: PDFMessage[], dateFormat: unknown, timeAndDateFormat: unknown): unknown[] {
```

Comment:
Why does this method have so many unknowns?

Reply:
This is because the data come from an external source and are uncertain. The `omnichannel-transcript` app doesn't share their types, so what we can do right now is set it as unknown...